### PR TITLE
libdeflate: register static build as test (ref: #144207)

### PR DESCRIPTION
**Pull request by KAction**
###### Motivation for this change

Fix static build of libdeflate

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KAction** [comment]
I know that static build is broken and I did not fix it yet. I want to make sure that with static build as test CI will fail.

File: pkgs/development/libraries/libdeflate/default.nix [hunk 1]
```diff
@@ -30,4 +30,5 @@ stdenv.mkDerivation rec {
     platforms = platforms.unix;
     maintainers = with maintainers; [ orivej ];
   };
+  passthru.tests.static = pkgsStatic.libdeflate;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libdeflate/default.nix:33]: passthru should be before meta.

**Comment by KAction** [comment]
Any idea why `libdeflate, libdeflate.passthru.tests on x86_64-linux — Failure` is marked as "neutral", not as failure?

File: pkgs/development/libraries/libdeflate/default.nix (hunk 1 above)
Comment (KAction) [reply pkgs/development/libraries/libdeflate/default.nix:33]: Ok, I'll fix.

**Comment by KAction** [comment]
I also added myself as maintainer to get notifications about changes. Do we have concept of "subscribe to particular path in repository"?

**Comment by SuperSandro2000** [comment]
> I also added myself as maintainer to get notifications about changes. Do we have concept of "subscribe to particular path in repository"?

Only GitHub Codeowners but I think you need commit permissions for that to work.

File: pkgs/development/libraries/libdeflate/default.nix [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+{ stdenv, lib, fetchpatch, fetchFromGitHub, fixDarwinDylibNames, pkgsStatic }:
 
-stdenv.mkDerivation rec {
+let
+  # Do it this way to prevent massive rebuild of reverse-dependencies
+  # of dynamic version. Should be inlined on next version update..
+  staticArgs = {
+    patches = [
+      (fetchpatch {
+        url = "https://github.com/ebiggers/libdeflate/pull/137.diff";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libdeflate/default.nix:9]:
```suggestion
        url = "https://github.com/ebiggers/libdeflate/commit/ee4d18872bfe09a32cfd031c716b9069a04a50a0.patch";
```

File: pkgs/development/libraries/libdeflate/default.nix [hunk 3]
```diff
@@ -23,11 +36,12 @@ stdenv.mkDerivation rec {
...
   meta = with lib; {
     description = "Fast DEFLATE/zlib/gzip compressor and decompressor";
     license = licenses.mit;
     homepage = "https://github.com/ebiggers/libdeflate";
     platforms = platforms.unix;
-    maintainers = with maintainers; [ orivej ];
+    maintainers = with maintainers; [ orivej kaction ];
   };
-}
+} // (if stdenv.hostPlatform.isStatic then staticArgs else { }))
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libdeflate/default.nix:47]: Can we just throw this onto staging to avoid this hackery?

Comment (KAction) [reply pkgs/development/libraries/libdeflate/default.nix]: Done.

**Comment by KAction** [comment]
What is the process to get commit bit?

**Comment by mkg20001** [comment]
If you're active enough you can apply through this issue https://github.com/NixOS/nixpkgs/issues/50105

**Comment by SuperSandro2000** [comment]
> If you're active enough you can apply through this issue 50105

Active is not a real hard requirement. I think the only requirements are that you are a trusted member of the community for a reasonable amount of time and you should use the commit bit somewhat regularly and don't be to inactive.

**Pull request merged**